src/utils/lst_gen.rs

```diff
@@ -17,49 +16,70 @@ pub fn update_or_create_status_list(
 
     let bits = bits.value() as usize;
 
-    let mut status_array: Vec<u8> = if let Some(existing_lst) = existing_lst {
-        // Decode & decompress existing list
-        let compressed_data = decode(&existing_lst).map_err(|_| Error::DecodeError)?;
-        let mut decoder = ZlibDecoder::new(&compressed_data[..]);
-        let mut decompressed = Vec::new();
-        decoder
-            .read_to_end(&mut decompressed)
-            .map_err(|e| Error::Generic(e.to_string()))?;
-        decompressed
-    } else {
-        // Determine the highest index to set the size of the status array
-        let max_index = status_updates
-            .iter()
-            .map(|update| update.index)
-            .max()
-            .ok_or_else(|| Error::Generic("Failed to determine max index".to_string()))?;
-
-        if max_index < 0 {
-            return Err(Error::InvalidIndex);
-        }
+    let max_index = status_updates
+        .iter()
+        .map(|update| update.index)
+        .max()
+        .ok_or_else(|| Error::Generic("Failed to determine max index".to_string()))?;
```

Comment:
I don't think returning an error when not finding the max index is appropriate here. You could instead assume the max index is `0`.
```suggestion
        .unwrap_or(0);
```

Reply:
No we can not do this because returning `0` will mean we found a token which is not true cause required len will be `1`